File: 7-kyu/mirroring-cipher/python/solution.py
def mirror(code, flip='abcdefghijklmnopqrstuvwxyz'):
    lookup_table = {}
    for i in range(len(flip)):
        lookup_table[flip[i]] = flip[len(flip) - i - 1]
    ret = ''
    for ch in code.lower():
        if ch in lookup_table:
            ret += lookup_table[ch]
        else:
            ret += ch
    return ret

File: 7-kyu/mirroring-cipher/javascript/solution.js
const mirror = (code, key = "abcdefghijklmnopqrstuvwxyz") => {
  const lookup = {};
  for (let i = 0; i < key.length; i++) {
    lookup[key[i]] = key[key.length - i - 1];
  }
  return code.toLowerCase().split("").map(ch => lookup[ch] || ch).join("");
};
